clio-server/src/main/scala/org/broadinstitute/clio/server/service/ServerService.scala

```diff
@@ -18,22 +17,30 @@ class ServerService private (
   httpServerDAO: HttpServerDAO,
   persistenceDAO: PersistenceDAO,
   searchDAO: SearchDAO
-)(implicit executionContext: ExecutionContext)
-    extends StrictLogging {
+)(implicit executionContext: ExecutionContext) {
 
   /**
-    * Kick off a startup, and return immediately.
+    * Kick off a startup, initializing all DAOs.
     */
-  def beginStartup(): Unit = {
-    val startupAttempt = startup()
-    startupAttempt.failed foreach { exception =>
-      logger.error(
-        s"Server failed to startup due to ${exception.getMessage}",
-        exception
+  def beginStartup(): Future[Unit] = {
+    for {
+      _ <- serverStatusDAO.setStatus(ServerStatusInfo.Starting)
+      _ <- persistenceDAO.initialize(
+        ElasticsearchIndex.WgsUbam,
+        ElasticsearchIndex.Gvcf
       )
-      shutdown()
-    }
-    ()
+      _ <- searchDAO.initialize()
+    } yield ()
+  }
+
+  /**
+    * Bind a port, and mark the server as ready to receive messages.
+    */
+  def completeStartup(): Future[Unit] = {
```

Comment:
Why is it preferable to split the startup into two different methods? In general we'd want to avoid having order dependencies on methods whenever possible.

Reply:
On startup, we want to:

1. Marks server as starting up
2. Initialize all DAOs
3. Recover metadata
4. Bind a port
5. Mark the server as started

1, 2, 4, and 5 all live in the `ServerService`, but 3 (right now) is in the `PersistenceService`. I can un-split the startup logic by either:
1. Moving the "recover" method into the `ServerService`, or
2. Passing a `PersistenceService` as an argument to the `ServerService`

I'd lean towards option 1, WDYT?

Reply:
1 sounds better to me